### ISSUE TITLE
rename Array item key prop

### ISSUE THREAD

**Issue by zepatrik** [rjsf-team/react-jsonschema-form#1428]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

The `ArrayField` component manages keys for it's items. These get passed down named `key` (see [here](https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/fields/ArrayField.js#L49)). Therefore they are not accessible inside custom array template items. Because of that you have to map/construct array items (also custom ones) like [this](https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/fields/ArrayField.js#L126). Renaming `key` to `itemKey`/`uid`/... would make it easier to write own components.

#### Expected behavior

I would expect to have access to the key over the props with usual react patterns.

#### Actual behavior

You have to use the mentioned workaround. If you want to have class components it gets even trickier.

### Version

master

**Comment by epicfaace** [comment]
Good point. The reactjs documentation also supports this: https://reactjs.org/warnings/special-props.html

**Comment by heath-freenome** [comment]
@zepatrik With the new templating capabilities in 5.x you may be able to control this better. Can you let us know if this is still an issue?

**Comment by heath-freenome** [comment]
The `key` prop was renamed to `itemKey` in this #4808
